plugins/modules/accesspoint_workflow_manager.py

```diff
@@ -0,0 +1,818 @@
+        Parameters:
+          self (object): An instance of a class used for interacting with Cisco Catalyst Center.
+          config (dict): A dictionary containing configuration information.
+        Returns:
+            self (object): An instance of a class used for interacting with Cisco Catalyst Center.
+        Description:
+            This method determines whether to update or create AP configuration in Cisco
+            Catalyst Center based on the provided configuration information. 
+            If the specified site exists, the method checks if it requires an update
+            by calling the 'update_ap_configuration' method. If an update is required, 
+            it calls the 'configure_access_points' function from the 'wireless' family of 
+            the Cisco Catalyst Center API. If Current configuration same as input configuration 
+            does not require an update, the method exits, indicating that Accesspoint 
+            configuration is up to date.
+        """
+
+        config_updated = False
+        config_created = False
+        task_response = None
+        # check if the given AP config exists and/or needs to be updated/created.
+
+        if self.have.get("device_exists"):
+            consolidated_data = self.compare_ap_cofig_with_inputdata(self.have["current_ap_config"])
+            if consolidated_data:
+                self.log('Final AP Configuration data to update {}'.format(str(consolidated_data)),
+                      "INFO")
+                task_response = self.update_ap_configuration(consolidated_data)
+                self.log('Task respoonse {}'.format(str(task_response)),"INFO")
+                config_updated = True
+            else:
+                # Accesspoint does not need update
+                self.msg = "AP - {0} does not need any update"\
+                    .format(self.have.get("current_ap_config").get("ap_name"))
+                self.log(self.msg, "INFO")
+                responses = {}
+                del self.payload["device_list"]
+                responses["accesspoints_updates"] = {"response": self.payload["device_config"]}
+                self.result['msg'] = self.msg
+                self.result["response"].append(responses)
+                self.result["skipped"] = True
+                return self
+        else:
+            #need to add the method for creation.
+            pass
+
+        if config_updated or config_created:
+            responses = {}
+            if task_response and isinstance(task_response, dict):
+                self.check_task_response_status(task_response, "task_intent", True).check_return_status()
+                if self.status == "success":
+                    self.result['changed'] = True
+                    responses["accesspoints_updates"] = {"response": task_response}
+                else:
+                    self.module.fail_json(msg="Unable to get Task Details.", response=task_response)
```

Comment:
Instead of raising exception, give appropriate failed msg - 
```
self.status = "failed"
self..msg = "Unable to get task response"
return self

Reply:
Changed.